test/testcafe/spec/IdentifyRecovery_spec.js

```diff
@@ -0,0 +1,93 @@
+import { RequestMock, RequestLogger } from 'testcafe';
+import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
+import xhrIdentifyRecovery from '../../../playground/mocks/data/idp/idx/identify-recovery';
+import xhrErrorIdentify from '../../../playground/mocks/data/idp/idx/error-identify-access-denied';
+import xhrAuthenticatorVerifySelect from '../../../playground/mocks/data/idp/idx/authenticator-verification-select-authenticator';
+
+const identifyRecoveryMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecovery)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRecoveryErrorMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecovery)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrErrorIdentify, 403);
+
+const identifyRequestLogger = RequestLogger(
+  /idx\/identify/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+fixture('Identify Recovery - reset flow');
+
+async function setup(t) {
+  const identityPage = new IdentityPageObject(t);
+  await identityPage.navigateToPage();
+
+  const { log } = await t.getBrowserConsoleMessages();
+  await t.expect(log.length).eql(3);
+  await t.expect(log[0]).eql('===== playground widget ready event received =====');
+  await t.expect(log[1]).eql('===== playground widget afterRender event received =====');
+  await t.expect(JSON.parse(log[2])).eql({
+    controller: 'forgot-password',
+    formName: 'identify-recovery',
+  });
+
+  return identityPage;
+}
+
+test.requestHooks(identifyRecoveryMock)('should have correct display texts', async t => {
+  // i18n values can be tested here.
+  const identityPage = await setup(t);
+
+  const identityPageTitle = identityPage.getPageTitle();
+  await t.expect(identityPageTitle).eql('Reset your password');
+
+  const saveButtonText = identityPage.getSaveButtonLabel();
+  await t.expect(saveButtonText).eql('Next');
+
+  await t.expect(await identityPage.signoutLinkExists()).ok();
+});
+
+test.requestHooks(identifyRequestLogger, identifyRecoveryMock)('should be able to submit identifier', async t => {
+  const identityPage = await setup(t);
+
+  await identityPage.fillIdentifierField('Test Identifier');
```

Comment:
nit: this should probably not have spaces in it right? since it is either an email or a username